vm/src/api.rs

```diff
@@ -1320,6 +1320,36 @@ macro_rules! record {
     }
 }
 
+#[macro_export]
+macro_rules! hlist_p {
```

Comment:
That is an internal macro but since it is used by the exported `record_p` macro it needs to be exported as well. Maybe I can hide it inside `record_p` instead.